frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/BasePushBranchAction.java

```diff
@@ -0,0 +1,66 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import java.util.Collections;
+
+import com.intellij.dvcs.push.ui.VcsPushDialog;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.project.Project;
+import git4idea.GitLocalBranch;
+import git4idea.push.GitPushSource;
+import git4idea.repo.GitRepository;
+import io.vavr.collection.List;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
```

Comment:
Nit: technically, this action doesn't require either of those keys since they're provided to `doPush` from outside; it's derived actions that likely require them

Reply:
good point!